sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/DoNotDecreaseMemberVisibility.cs

```diff
@@ -0,0 +1,198 @@
+                    if (hidingProperty != null)
+                    {
+                        var location = (memberDeclaration as PropertyDeclarationSyntax).Identifier.GetLocation();
+                        return Diagnostic.Create(rule, location, hidingProperty);
+                    }
+                }
+                return null;
+            }
+
+            private static bool DecreasesPropertyAccess(IPropertySymbol baseProperty, IPropertySymbol propertySymbol)
+            {
+                if (baseProperty.Name != propertySymbol.Name ||
+                    !Equals(baseProperty.Type, propertySymbol.Type))
+                {
+                    return false;
+                }
+
+                var baseGetAccess = GetEffectiveDeclaredAccess(baseProperty.GetMethod, baseProperty.DeclaredAccessibility);
+                var baseSetAccess = GetEffectiveDeclaredAccess(baseProperty.SetMethod, baseProperty.DeclaredAccessibility);
+
+                var propertyGetAccess = GetEffectiveDeclaredAccess(propertySymbol.GetMethod, baseProperty.DeclaredAccessibility);
+                var propertySetAccess = GetEffectiveDeclaredAccess(propertySymbol.SetMethod, baseProperty.DeclaredAccessibility);
+
+                return DecrasesAccess(baseGetAccess, propertyGetAccess) ||
+                       DecrasesAccess(baseSetAccess, propertySetAccess);
+            }
+
+            private static Accessibility GetEffectiveDeclaredAccess(IMethodSymbol method, Accessibility propertyDefaultAccess)
+            {
+                if (method == null)
+                {
+                    return Accessibility.NotApplicable;
+                }
+
+                return method.DeclaredAccessibility != Accessibility.NotApplicable ? method.DeclaredAccessibility : propertyDefaultAccess;
+            }
+
+            private static bool IsMatchingSignature(IMethodSymbol baseMethod, IMethodSymbol methodSymbol)
+            {
+                if (baseMethod.Name != methodSymbol.Name ||
+                    baseMethod.TypeParameters.Length != methodSymbol.TypeParameters.Length)
+                {
+                    return false;
+                }
+
+                bool hasMatchingParameterTypes = CollectionUtils.AreEqual(baseMethod.Parameters, methodSymbol.Parameters,
+                    AreParameterTypesEqual);
+
+                return hasMatchingParameterTypes;
+            }
+
+            private static bool AreParameterTypesEqual(IParameterSymbol p1, IParameterSymbol p2)
+            {
+                if (p1.Type.TypeKind == TypeKind.TypeParameter)
+                {
+                    return p2.Type.TypeKind == TypeKind.TypeParameter;
+                }
+
+                return Equals(p1.Type.OriginalDefinition, p2.Type.OriginalDefinition);
+            }
+
+            private static bool DecrasesAccess(Accessibility baseAccess, Accessibility memberAccess)
```

Comment:
Same remark for the name of the method. Besides you made a typo in the name (Decrases instead of Decreases). There is a spell checker extension for VS you might want to use.